test/tests/pbs_smoketest.py

```diff
@@ -79,15 +79,38 @@ def test_submit_job_array(self):
 
     def test_advance_reservation(self):
         """
-        Test to submit an advanced reservation
+        Test to submit an advanced reservation and submit jobs to that
+        reservation. Check if the reservation gets confimed and the jobs
+        inside the reservation starts running when the reservation runs.
         """
-        r = Reservation()
-        a = {'Resource_List.select': '1:ncpus=1'}
+        a = {'resources_available.ncpus': 4}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+        r = Reservation(TEST_USER)
+        now = int(time.time())
+        a = {'Resource_List.select': '1:ncpus=4',
+             'reserve_start': now + 5,
+             'reserve_end': now + 105}
         r.set_attributes(a)
         rid = self.server.submit(r)
         a = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
         self.server.expect(RESV, a, id=rid)
 
+        # submit a normal job and an array job to the reservation
+        a = {'Resource_List.select': '1:ncpus=1',
+             ATTR_q: rid.split('.')[0]}
+        j1 = Job(TEST_USER, attrs=a)
+        jid1 = self.server.submit(j1)
+
+        a = {'Resource_List.select': '1:ncpus=1',
+             ATTR_q: rid.split('.')[0], ATTR_J: '1-2'}
```

Comment:
Since you're splitting the rid twice, why not do it once and store it into a local variable?